src/server/editorServices.ts

```diff
@@ -1451,6 +1484,7 @@ namespace ts.server {
             // at this point if file is the part of some configured/external project then this project should be created
             const info = this.getOrCreateScriptInfoForNormalizedPath(fileName, /*openedByClient*/ true, fileContent, scriptKind, hasMixedContent);
             this.assignScriptInfoToInferredProjectIfNecessary(info, /*addToListOfOpenFiles*/ true);
+            this.deleteOrphanScriptInfoNotInAnyProject();

```

Comment:
please add a comment here on why we do this on open, just to make sure ppl do not wonder in the future why this is here and not in close project.